### DISCUSSION TITLE
Supabase OAuth in Unity

### DISCUSSION THREAD

**Discussion by gaetandezeiraud** [supabase-community/supabase-csharp#154] [Q&A] [answered]
Hi!
I tried to setup OAuth sign in, in a Unity project. But I encounter some problems. To be more specific, one problem, the fact that the access_token is in the Fragment part of the URL.

More here with the code https://github.com/wiverson/supabase-unity-template/issues/6

Any idea how to do? Maybe my approach is not the good one. Any help is appreciated 😅.

Thanks.

**Comment by wiverson** [comment, marked as answer]
The short answer is that you would have to open up a web view to start the flow, and then have a way to get the completed flow back to the native app. It's doable, but it's non-trivial. You'll need to coordinate between your Unity app, a custom web app you will have to write, and then have a way to get the session info back again from the custom web app after the flow is completed with the OAuth provider. This could be done by having some kind of WebSocket or polling back to the server from the Unity client.

You can see an example of how a flow like this works eg with the JetBrains toolbox when you log in, except that the flow would also have to include kicking over to the 3rd party OAuth provider as well as your own web server.

Technically you have all of the tooling you need to build this between the Supabase JS client for the custom web app and the C# lib for the Unity project. But it's not just a oneliner login or anything.

**Comment by gaetandezeiraud** [comment]
A solution here https://github.com/wiverson/supabase-unity-template/discussions/7#discussioncomment-9302150
It is not the best fit. But works.
